Nextjs v13+ sharp package not included issue

### Current Behavior

Currently when deploying `nextjs` to `production` and leveraging the `image optimization` it is recommended to include `sharp` as a dependency. However, this dependency is `not referenced anywhere` in the `users nx monorepo source code` and thus - `not included in the dependency graph` when building nextjs application which means - generated `production package.json does not include the required dependency`.

The same issue is also for `experimental.optimizeCss` option which requires `critters` package as a dependency.


### Expected Behavior

When adding `sharp` and `critters` dependency it should also include these packages into nextjs application build. 
Another alternative would be to enable to provide extra dependencies inside `project.json` of `nx/next` configuration of the `package.json` to include additional dependencies.

### GitHub Repo

_No response_

### Steps to Reproduce

1. Create an nx/next project
2. Use <Image /> component from `next/image`
3. Build for production
4. Start production server
5. Get this warning: ` ⚠ For production Image Optimization with Next.js, the optional 'sharp' package is strongly recommended. Run 'npm i sharp', and Next.js will use it automatically for Image Optimization. `
6. Add `sharp` as a dependency inside root `package.json`
7. Build for production
8. Start production server
9. Still get the same warning
10. Check `package.json` -> `sharp` not added.

`experimental.optimizeCss`
1. add an option at next.config.js `experimental: { optimizeCss: true }`
2. Build production
3. Start production
4. Get this Error when starting server: `Error: Cannot find module 'critters'`
5. Add `critters` as a dependency in root `package.json`
6. Build production
7. Start production
8. The same error occurs
9. Check `dist/apps/<app>/package.json` -> `critters` not added

**NOTE:** Serve production from inside `dist/apps/<app>` with `npm i` and `npm run start` performed.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 20.9.0
   OS     : win32-x64
   npm    : 9.5.1
   
   nx                 : 17.2.4
   @nx/js             : 17.2.4
   @nx/jest           : 17.2.4
   @nx/linter         : 17.2.4
   @nx/eslint         : 17.2.4
   @nx/workspace      : 17.2.4
   @nx/devkit         : 17.2.4
   @nx/eslint-plugin  : 17.2.4
   @nx/next           : 17.2.4
   @nx/react          : 17.2.4
   @nrwl/tao          : 17.2.4
   @nx/web            : 17.2.4
   typescript         : 5.2.2
```


### Failure Logs

```shell
> nx run next:serve:production

   ▲ Next.js 14.0.4
   - Local:        http://localhost:4200
 ✓ Ready in 293ms
 ⚠ For production Image Optimization with Next.js, the optional 'sharp' package is strongly recommended. Run 'npm i sharp', and Next.js will use it automatically for Image Optimization.
Read more: https://nextjs.org/docs/messages/sharp-missing-in-production

> nx run next:build:production

   ▲ Next.js 14.0.4
   - Experiments (use at your own risk):
     · optimizeCss
   Creating an optimized production build ...
 ✓ Compiled successfully
   Skipping linting
   Checking validity of types ...
   Collecting page data ...
   Generating static pages (0/5) ...
   Generating static pages (1/5) 
Error occurred prerendering page "/404". Read more: https://nextjs.org/docs/messages/prerender-error
Error: Cannot find module 'critters'
```


### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

## Current workaround:
```
const rootPackageJson = require('../package.json');
const nextBuildPackageJson = require('../dist/apps/web/package.json');
const { writeFileSync } = require('fs');

/**
 * Currently nx omits required dependencies from root package.json such as `sharp`, `critters`
 * This is a workaround to include this dependency into production build
 */
nextBuildPackageJson.dependencies.sharp = rootPackageJson.dependencies.sharp;
nextBuildPackageJson.dependencies.critters =
  rootPackageJson.dependencies.critters;

if (process.env.NODE_ENV !== 'production' && !process.env.CI) {
  nextBuildPackageJson.scripts.start =
    nextBuildPackageJson.scripts.start.concat(' -p 4200');
}

writeFileSync(
  __dirname + '/../dist/apps/web/package.json',
  JSON.stringify(nextBuildPackageJson, null, 4)
);

```

--- a/packages/next/src/executors/build/lib/update-package-json.ts
+++ b/packages/next/src/executors/build/lib/update-package-json.ts
@@ -15,7 +15,16 @@ export function updatePackageJson(
   packageJson.dependencies ??= {};
 
   // These are always required for a production Next.js app to run.
-  const requiredPackages = ['react', 'react-dom', 'next', 'typescript'];
+  // sharp is for next/image https://nextjs.org/docs/messages/sharp-missing-in-production
+  // critters is required for experimental optimizing CSS
+  const requiredPackages = [
+    'react',
+    'react-dom',
+    'next',
+    'typescript',
+    'sharp',
+    'critters',
+  ];
   for (const pkg of requiredPackages) {
     const externalNode = context.projectGraph.externalNodes[`npm:${pkg}`];
     if (externalNode) {

